prow/resultstore/writer/writer.go

```diff
@@ -81,10 +81,10 @@ type writer struct {
 	finalized   bool
 }
 
-// persistentError returns whether the error status code is not
+// PermanentError returns whether the error status code is not
```

Comment:
Consider adding "A PermanentError is classified as such because these errors are largely outside of our control, at least as far as how we interact with ResultStore" ?